### PR TITLE
Issue #485 Relay namecoin id to other nodes

### PR DESCRIPTION
**Pull request by charapod**
1) As requested in #957, we changed namecoin_id column to lowercase. 

But! Someone who had already upgraded their database with the old migration scripts will still have their column named with a capital, and there's no easy way to get rid of it since sqlite does not support DROP COLUMN (issue #955). New databases created will work fine of course, and rerunning the migration4.py script will probably do it as well. Let us know if you want us to cancel this modification.

2) We begun working on issue #485 by @dionyziz . We added the namecoin of the sender to the rest of the information in the 'start_handshake' and 'send' methods in CryptoPeerConnection and we also stored the namecoin the user entered with the rest of his info in CryptoTrasportLayer. Is this the right way to do it? Should we change anything else? We need a little feedback here @gubatron , @hoffmabc :) 

3) As it is now, TransportLayer does not have a 'namecoin' field. Should we add a namecoin field in there, and treat it in the same manner as nickname?

4)Should we add the namecoin to the 'ok_msg' in transport.py (line 136)?

5)We see that you sign the information *before* you encrypt it. Is that intended? afaik this is not secure...

### CODE REVIEW

File: node/trust.py [hunk 1]
```diff
@@ -39,3 +41,19 @@ def get_history():
...
+
+
+def is_valid_namecoin(namecoin, guid):
+    if not namecoin or not guid:
+        return False
+
+    server = DNSChainServer.Server(constants.DNSCHAIN_SERVER_IP, "")
+    _log.info("Looking up namecoin id: %s", namecoin)
+    try:
+        data = server.lookup("id/"+namecoin)
```
Comment (Renelvon) [inline node/trust.py]: To be fair, I can't see any PEP8 issue here, save for the spaces around `+`. Is that it?

Comment (dionyziz) [inline node/trust.py]: yep

File: requirements.txt [hunk 2]
```diff
@@ -15,3 +15,4 @@ requests==2.4.3
 tornado==4.0.2
 Twisted==14.0.2
 rfc3986==0.2.0
+git+https://github.com/okTurtles/pydnschain#egg=pydnschain
```
Comment (dionyziz) [inline requirements.txt]: Oh whoops, I'll get to it.

**Comment by charapod** [comment]
Happy New Year guys :D! Shall I remove the "WIP" flag? :P

File: node/trust.py (hunk 1 above)
Comment (lolosk) [inline node/trust.py]: Done! :)

**Comment by lolosk** [comment]
Fixed the style issue.

Happy new year! :D

**Comment by Renelvon** [comment]
> Happy New Year guys :D! Shall I remove the "WIP" flag? :P

Happy New Year! You are welcome to remove the `WIP` when you feel the PR is complete and ready for review.

File: requirements.txt (hunk 2 above)
Comment (dionyziz) [inline requirements.txt]: Pushed:
https://pypi.python.org/pypi/dnschain/0.1.0

Seems to work now:
```
dionyziz@erdos ~ % pip install dnschain
Downloading/unpacking dnschain
  Downloading dnschain-0.1.0.tar.gz
  Running setup.py (path:/private/var/folders/ty/nt50gpgx1737thg9ll863bj80000gn/T/pip_build_dionyziz/dnschain/setup.py) egg_info for package dnschain

Installing collected packages: dnschain
  Running setup.py install for dnschain

Successfully installed dnschain
Cleaning up...
dionyziz@erdos ~ % cd ..
dionyziz@erdos ~ % cd ..
dionyziz@erdos ~ % python
Python 2.7.8
>>> import dnschain
>>>
dionyziz@erdos ~ %
```

So, let's switch to pypi.

Comment (lolosk) [inline requirements.txt]: Done. Seems to work fine now.

**Comment by lolosk** [comment]
Moved dnschain installation to pypi. Thanks @dionyziz!

**Comment by dionyziz** [comment]
@charapod Great work folks, amazing to see this finally merged!

**Comment by dionyziz** [comment]
@charapod Have a drink on me tonight @changetip

**Comment by dionyziz** [comment]
@lolosk Have a drink on me tonight @changetip

**Comment by changetip** [comment]
Hi @charapod, I've delivered a Bitcoin tip worth a drink (18,678 bits/$5.04) from @dionyziz to your ChangeTip wallet.

**[Learn about ChangeTip](https://www.changetip.com/tip-online/github)**

**Comment by changetip** [comment]
Hi @lolosk, @dionyziz sent you a Bitcoin tip worth a drink (18,678 bits/$5.04), and I'm here to deliver it ➔ **[collect your tip](https://www.changetip.com/collect/324885)**.


**[Learn more about ChangeTip](https://www.changetip.com/tip-online/github)**

**Pull request merged**